### PR TITLE
Use Observable Subscription To Update Local Store As Messages Are Processed

### PR DESCRIPTION
**Pull request by johnhalbert**
Pull request for suggestion in issue #55.

Note: Using the types provided by `most`, type parameters are typed as `any`, which is obviously not good.  It seems to me the proper place for that type would be from the `StreamDSL`, and ultimately the `KTable` class.  Something like:

```
class StreamDSL<K, V> implements Observable<KafkaMessage<K, V>> {
 . . .
}
```
And
```
class KTable<K, V> extends StreamDSL<K, V> {
 . . .
}
```

But, that was really beyond the scope of the suggestion, so I've left it with type `any` for now, which follows with the current implementations typing.  In any case, should be easy to upgrade if types get added to those classes.

### CODE REVIEW

File: src/lib/KStorage.ts [hunk 1]
```diff
@@ -76,6 +78,11 @@ export class KStorage {
 	  return Promise.resolve(true);
 	}
 
+  start(subscription: Subscription<any>) {
```
Comment (rob3000) [inline src/lib/KStorage.ts:81]: Missing return types and can we include doc block comments too please.

File: src/lib/KStorage.ts [hunk 2]
```diff
@@ -85,8 +92,22 @@ export class KStorage {
...
+  next({ key, value }) {
+    return this.set(key, value);
+  }
+
+  error(error) {
+    // Not much to do in this context.
+    console.error(error);
+  }
+
+  complete() {
```
Comment (rob3000) [inline src/lib/KStorage.ts:108]: Missing return type and doc block.

File: src/lib/KStorage.ts [hunk 3]
```diff
@@ -85,8 +92,22 @@ export class KStorage {
 	}
 
 	close() {
+    this._subscription.unsubscribe();
 	  return Promise.resolve(true);
 	}
+
+  next({ key, value }) {
```
Comment (rob3000) [inline src/lib/KStorage.ts:99]: Indentation looks out here, and missing return type and doc block comments

**Review by rob3000 — CHANGES_REQUESTED** [review]
Just a few comments too look at. Also looks like the checks are failing due too:
```
TypeError: Cannot read property 'unsubscribe' of undefined
at KStorage.close (src/lib/KStorage.ts:95:24)
```

File: src/lib/KStorage.ts (hunk 1 above)
Comment (johnhalbert) [reply src/lib/KStorage.ts]: Added these

File: src/lib/KStorage.ts (hunk 2 above)
Comment (johnhalbert) [reply src/lib/KStorage.ts]: Added these

File: src/lib/KStorage.ts (hunk 3 above)
Comment (johnhalbert) [reply src/lib/KStorage.ts]: Added doc / return type. Using vim, I'm not seeing the same indentation issue as shown on GitHub. Could you check it locally as well?

**Comment by johnhalbert** [comment]
@rob3000 sorry for taking so long on this.  Made requested changes, fixed failing tests.

**Comment by jonahharris** [comment]
This has been sitting for a while and it would be nice to have a streaming KTable for my project.

**Comment by johnhalbert** [comment]
@rob3000 any chance you could take a look at this?

**Review by rob3000 — APPROVED** [review]

**Pull request merged**